File: 23Jan2024.java
import java.util.List;

class Solution {
    public int maxLength(List<String> arr) {
        int ans = 0;
        for (int i = 1; i < (1 << arr.size()); i++) {
            boolean isValid = true;
            int l = 0;
            int[] fre = new int[26];
            for (int j = 0; j < arr.size() && isValid; j++) {
                if ((i & (1 << j)) != 0) {
                    for (int k = 0; k < arr.get(j).length(); k++) {
                        if (fre[arr.get(j).charAt(k) - 'a'] > 0)
                            isValid = false;
                        fre[arr.get(j).charAt(k) - 'a']++;
                    }
                    // System.out.println()
                    l += arr.get(j).length();
                }
            }
            // System.out.println(i+" "+l+" "+isValid);
            if (isValid)
                ans = Math.max(ans, l);
        }
        return ans;
    }
}

File: 23Jan2024.py
from ast import List


class Solution:
    def maxLength(self, arr: List[str]) -> int:
        ans = 0
        fre = [0]*26
        for i in range(1, (1 << len(arr))):
            isValid = True
            l = 0
            for j in range(0,26) :
                fre[j]=0
            for j in range(len(arr)):
                if (i & (1 << j)) != 0:
                    for k in range(len(arr[j])):
                        if fre[ord(arr[j][k]) - ord('a')] > 0:
                            isValid = False
                        fre[ord(arr[j][k]) - ord('a')] += 1
                    l += len(arr[j])
            if isValid:
                ans = max(ans, l)
        return ans